final_task/rss_reader/output_functions.py

```diff
@@ -0,0 +1,355 @@
+              "Main source: {7}" \
+        .format(item.number_of_novelty,
+                pprint.pformat(item.title_of_novelty, width=115),
+                item.time_of_novelty,
+                pprint.pformat(item.source_link, width=115),
+                pprint.pformat(item.description.replace("\xa0", " "), width=115),
+                pprint.pformat(item.images_links),
+                pprint.pformat(item.alt_text, width=115),
+                item.main_source)
+    logging.info("Got novelty into file!")
+    return novelty
+
+
+def getting_corrected_time(item):
+    """
+    Getting time in view %Y%m%d
+    """
+    logging.info("Getting corrected date!")
+    corrected_date = datetime.strptime(item.get('published', ''), '%a, %d %b %Y %X %z')
+    logging.info("Got corrected date!")
+    return corrected_date.strftime('%Y%m%d')
+
+
+def getting_time_for_json(item):
+    corrected_date = datetime.strptime(item, '%a, %d %b %Y %X %z')
+    return corrected_date.strftime('%Y%m%d')
+
+
+def reading_file(name_of_file):
+    with open(name_of_file, 'r', encoding='utf-8') as news_cache:
+        return news_cache.read()
+
+
+def writing_to_file(pack_of_news, pack_of_news_for_db, filename):
+    """
+    Writing information into 2 files: news_cache.txt and News_cache_json.json
+    Creating 2 files because it's easier to read information into computer from JSON file than another file
+    1. Open file
+    2. Check if file is empty or not
+    3. If empty - append whole information
+       If Not empty:
+              1. Check if the novelty in file
+              2. If in file: continue
+                 If Not in the file: Find out length of file (amount of news), put that number to incoming novelty
+    If that novelty exists it will go to another novelty in limit
+    It means that if you enter --limit 15 and 10 news are already in list it will add only 5 news to list
+    """
+    conn = sqlite3.connect("database.db")
+    cursor = conn.cursor()
+    cursor.execute('create table if not exists projects(num integer, title text, time text, source_link text, '
+                   'description text, images_links text, alt_tx text, date_corrected integer, main_source text)')
+    conn.commit()
+    logging.info("Opening file News_cache.")
+    with open(filename, 'a', encoding='utf-8') as news_cache:
+        logging.info("Reading from News_cache.")
+        content = reading_file('news_cache.txt')
+        if not content:
+            logging.info("Writing news to an empty file!")
+            writing_if_file_empty(news_cache, pack_of_news, pack_of_news_for_db)
+            logging.info("Wrote news to an empty file!")
+        else:
+            logging.info("Writing news to NOT empty file!")
+            writing_if_something_in_file(news_cache, pack_of_news, pack_of_news_for_db, content)
+            logging.info("Wrote news to NOT empty file!")
```

Comment:
If you are storing news in the database, do you really need text cache file?

Reply:
I use it to make it easier for you to read news